src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4938,7 +4960,7 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
 
   if ( !context->hasCachedValue( cacheIndex ) )
   {
-    spatialIndex = QgsSpatialIndex( cachedTarget->getFeatures() );
+    spatialIndex = QgsSpatialIndex( cachedTarget->getFeatures(), nullptr, QgsSpatialIndex::FlagStoreFeatureGeometries );
```

Comment:
Is this for the nearest neighbor function?

Reply:
Yes I looked into new join by nearest processing algoritm code and I found it: https://github.com/qgis/QGIS/blob/8416e7588c4f05d3c69f792e0c0ef4589084e49d/src/analysis/processing/qgsalgorithmjoinbynearest.cpp#L189
It is for indexing full geometry instead of their boundary

Reply:
I would suggest we either do this conditionally in here or make a new `nearestNeighbourOverlay` function.
Anyway, let's defer any work on nearest neighbour until everything else is in place, the nearest neighbour implementation is different enough from the others to have some special treatment, and I would like to try to avoid any distraction for the base functionality.

Reply:
okkk